### ISSUE TITLE
CI is very slow

### ISSUE THREAD

**Issue by eschnett** [openPMD/openPMD-api#1538]
I noticed that the CI builds are very slow. Each build starts with building cmake and related tools which takes about 20 minutes.

One way out would be to install these via Ubuntu (or to download a binary) and use `spack external find` to detect these tools. Another way would be to produce and use Docker images for the various CI setups.

**Comment by ax3l** [comment]
Agreed. Unfortunately there is not enough binary coverage in Spack binary caches yet to cover all our CI.

But adding more externals here:
- https://github.com/openPMD/openPMD-api/tree/dev/.github/ci/spack-envs

and reworking our binary cache logic as we did here:
- https://github.com/AMReX-Codes/pyamrex/pull/188

Is needed to speed things up again.

**Comment by ax3l** [comment]
Taking note of new spack caching possibilities:
https://github.com/spack/spack/releases/tag/v0.21.0
